app/controllers/concerns/authentication.rb

```diff
@@ -14,16 +14,18 @@ def authenticate_user!
 
   def login(user)
     reset_session
-    session[:current_user_id] = user.id
+    user.regenerate_session_token
+    session[:current_user_session_token] = user.reload.session_token
   end
 
   def forget(user)
     cookies.delete :remember_token
     user.regenerate_remember_token
   end
 
-  def logout
+  def logout(user)
```

Comment:
Rails nitpick: don't you have access to `current_user` here, so you don't need to pass it? I assume you're always signing out the current user.

Reply:
@mike-burns thank you for pointing this out. I just made this update, but it meant that I needed to call it before calling `reset_session`. This is because `current_user` returns `nil` after `reset_session` is called.

Do you think it's less secure to reset the session after regenerating the token? Could there ever be a case where calling `current_user.regenerate_session_token` fails and then `reset_session` is never reached?

Reply:
Ah I see.

Well you could also do,

```ruby
user = current_user
reset_session
user.regenerate_session_token
```

As you point out, DB things can fail, and you want to make sure you reset the session regardless.